contracts/EthProxy.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.6.7;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "./IUSM.sol";
+import "./external/IWETH9.sol";
+
+
+contract EthProxy {
+    using Address for address payable;
+    IUSM public usm;
+    IWETH9 public weth;
+
+    constructor(address usm_, address weth_) public {
+        usm = IUSM(usm_); 
+        weth = IWETH9(weth_);
+
+        weth.approve(address(usm), uint(-1));
+    }
+
+    /// @dev The WETH9 contract will send ether to EthProxy on `weth.withdraw` using this function.
+    receive() external payable { }
```

Comment:
Can you explain in like one line (just to me, not nec in the code) how this empty-body fn does anything?

Reply:
Note that it is not `function receive()` but `receive()`.

Until solidity 0.6 the fallback function implemented the logic to be executed when receiving ether. In solidity 0.6 and later the logic to execute when receiving ether must be included in the special `receive()` function.

If the body of `receive()` is empty, it just does the standard fare, updating balances and whatever.

`receive()` needs to exist in `EthProxy.sol`, so that it can receive ether from WETH9 on executing `weth.withdraw()`

Reply:
Ah yeah OK I did read about `receive()` vs `fallback()` at some point but I forgot about [this part](https://solidity.readthedocs.io/en/latest/contracts.html#receive-ether-function): "If neither a receive Ether nor a payable fallback function is present, the contract cannot receive Ether through regular transactions and throws an exception."  Got it now, thanks.